controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -162,6 +156,22 @@ type NetworkSpec struct {
 	NetworkType string `json:"networkType,omitempty"`
 }
 
+// ClusterComputeSpec defines the configuration for the required worker nodes provisioned as part of the cluster creation.
+type ClusterComputeSpec struct {
```

Comment:
Could we call this struct something different then? For example `defaultNodePool` and specify why/when this configuration is needed? are we also planning to surface this information as a MachinePool?

Reply:
I didn't want to call it `defaultNodePool` since OCM can create multiple nodepools (one per az).
the rosa cli flag for this is called `computeMachineType`, so I thought this will be less confusing for customer transition from the cli, wdyt?

> specify why/when this configuration is needed

`InstanceType` and `Autoscaling` are optional, only the `availabilityZones` field is required, we could probably infer this from the `subnetIDs` field in the future.

> are we also planning to surface this information as a MachinePool?

At first I was planing to do that when I thought only 1 default nodepool will be created, but given the fact that multiple default nodepools can be created. it didn't seem like a good idea anymore. the logic is more complicated and the resulting `ROSAMachinePool` CRs will be different from CRs created manually because OCM limits what fields you can edit on the defaul machinepools and also prevents you from deleting them.

Reply:
> What's the purpose of these configuration field? My understanding was that ROSA uses these fields to create a default pool for each availability zone?

correct

> Is there an issue open for it?

https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4459

> Do we want to move availabilityZones outside given that's a required field?
> If we don't surface them, users would incur in additional costs hidden away from their configuration, which isn't ideal.

since its difficult to surface those default machinepools today, my idea was to have this required field there, so you would always get the `computeSpec` field in your manifest so you know that you are getting compute resources  created per az you specify. 
This is not great, but the rosa cli will create the same default machinepools when you do `rosa create cluster --avaialbity-zones az1, az2 ...`

The plan is for OCM/ROSA to stop creating default machinePools altogether. Then we would just need to drop this `clusterCompute` field in CAPI.